canvas/rectangle.go

```diff
@@ -12,6 +12,10 @@ var _ fyne.CanvasObject = (*Rectangle)(nil)
 // Rectangle describes a colored rectangle primitive in a Fyne canvas
 type Rectangle struct {
 	baseObject
+	// Support shadow configuration
```

Comment:
Embedding at the beginning of the struct like this will break any code that has done `&Rectangle{color.Transparent}` - though that is not best practice we try to avoid breaking if we can.